to-do-list/src/components/Task.js

```diff
@@ -0,0 +1,41 @@
+import { useState } from "react";
+import Button from "./Button"
+
+function Task(props) {
+
+    const { task, date, status, completeTask, curDate } = props
+
+    const dueDate = new Date(date) 
+
+    let difference = dueDate - curDate; //subtract Due Date with Current Date
+
+    const differenceDays = Math.floor(difference / (1000 * 60 * 60 * 24)) //Days
+
+    const differenceTime = Math.floor((difference / (1000 * 60 * 60)) % 24)//Time
+
+    let lateDue = curDate - dueDate; 
+
+    const lateDays = Math.floor(lateDue / (1000 * 60 * 60 * 24))
+
+    const lateTime = Math.floor((lateDue / (1000 * 60 * 60)) % 24)
```

Comment:
All of this date functionality would fit best in a separate function in a separate helper file. It's always a good practice to separate your logic from your components whenever possible.